### ISSUE TITLE
indent: add option for multiline [assignment] statement

### ISSUE THREAD

**Issue by jsejcksn** [eslint/eslint#12248]
**What rule do you want to change?**
[indent](https://eslint.org/docs/rules/indent)

**Does this change cause the rule to produce more or fewer warnings?**
More

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option (and hopefully default behavior 🤞)

**Please provide some example code that this change will affect:**

```js
const getRepos = async (req, res) => {
  try {
    const fetchRes = await fetch(`https://api.github.com/users/${req.params.user}/repos`);
    const json = await fetchRes.json();
    res.json(json);
  }
  catch (err) {
    const statusCode = 500;
    res.status(statusCode).send(err.message);
  }
};
```

And line 3 is the specific one in question:

```js
    const fetchRes = await fetch(`https://api.github.com/users/${req.params.user}/repos`);
```

Its length is greater than 80 characters, so I need to wrap the line. I can wrap it in any of the following ways without `eslint` complaining:

```js
    const fetchRes =
      await fetch(`https://api.github.com/users/${req.params.user}/repos`);
```

```js
    const fetchRes =
          await fetch(`https://api.github.com/users/${req.params.user}/repos`);
```

```js
    const fetchRes =
await fetch(`https://api.github.com/users/${req.params.user}/repos`);
```

But I would only like to allow the first way to be valid.

**Are you willing to submit a pull request to implement this change?**
Not no, but also not yes. I've never looked at this codebase.

**Comment by kaicataldo** [comment]
This seems like a reasonable enhancement to me, though I'm not sure what we should call the option name. Thoughts? Maybe `multilineAssignment`?

**Comment by jsejcksn** [comment]
If this ends up being specific to assignments, then I think that's a great name. If it might apply to other types of statements or expressions when implemented (see #12255), then perhaps it should be something like `multilineExpression`.

**Comment by kaicataldo** [comment]
`MultilineExpression` seems like a better plan to me, personally, but I’m curious what the rest of the team thinks.

**Comment by golopot** [comment]
If the option name is `MultilineExpression`, if will also need to handle binary expressions:
```js
if (
  a && (
    a.b ||
    a.c
  ) &&
  a.d
) {}

if (
  aaaaaa >
bbbbb
) {}

const a = 
 x +
    y * z


```

**Comment by kaicataldo** [comment]
I'll champion this.

**Comment by Zamiell** [comment]
@kaicataldo Thanks for championing this proposal! It would be a tremendous boon to me and I'm sure plenty of others as well, as demonstrated in #12674, #12945, #13153, etc.

I'm noticing that there hasn't been any activity on this thread in a while. Has there been problems in getting consensus from three team members?

**Comment by kaicataldo** [comment]
@eslint/eslint-team Would anyone else like to support this proposal? If not, I think it's unfortunately time to close this issue.

**Comment by jsejcksn** [comment]
@kaicataldo In this context, what is the meaning of “support this proposal”?

**Comment by kaicataldo** [comment]
Our definition of reaching consensus is outlined [here](https://eslint.org/docs/maintainer-guide/issues#accepting-issues). In this case, we have a champion and one supporter from the team. We need two more supports from the team before we can accept this proposal.

**Comment by jsejcksn** [comment]
@kaicataldo Thanks for the link.

One thing I don’t see in the content is this: how are team members made aware of proposals which still need consensus? Are they notified somehow? Or might they just miss certain proposals if they don’t search for them?

**Comment by kaicataldo** [comment]
It's on the champion to try to seek consensus. We have a very small team made up mostly of volunteers who work on the project in their free time, and this process is intended to allow us to prioritize proposals that the team deems highest priority. This does mean that many proposals aren't accepted (even if some team members support them), but it's what allows us to keep the project alive.

**Comment by Tayflo** [comment]
If I may say, this would seem very useful to me (simple eslint user) since it would be a _wide-range_ ruling option, concerning multiples operators (assignment, arithmetic, logical, string, ternary...), hence a lot of statements (especially declarations and condition expressions).

The absence of such a rule allows writing of nasty (and inconsistent) code without warning, and it seems quite unavoidable when using the `max-len` rule.

**Comment by nzakas** [comment]
@kaicataldo how do you want to proceed here? According to our [new rule policy](https://eslint.org/blog/2020/05/changes-to-rules-policies) this change would be excluded, but since you championed this before the announcement we can make an exception if you have a plan for it.

**Comment by kaicataldo** [comment]
It's been 14 days since I left [this comment](https://github.com/eslint/eslint/issues/12248#issuecomment-628819305) and we still haven't been able to reach consensus, so I think it's time to close it. 

Thanks for the discussion, all!

**Comment by LiandriCorp** [comment]
@kaicataldo is there a protocol for revisiting issues? 😃 

My project has a lot of "complex" logic, and it seems we can't use eslint to enforce simplifying the steps (many named variables, minimum X conditions per variable), nor to organize the steps via indentation. Or maybe a new rule option has popped up I haven't noticed yet? Thanks, all.
